application.go

```diff
@@ -431,12 +429,14 @@ func (r *marathonClient) WaitOnApplication(name string, timeout time.Duration) e
 			flick.SwitchOn()
 		}()
 		for !flick.IsSwitched() {
-			if found, err := r.HasApplication(name); err != nil {
-				continue
-			} else if found {
-				if app, err := r.Application(name); err == nil && app.AllTaskRunning() {
-					return nil
+			if err := r.HasApplication(name); err != nil {
+				if err != ErrDoesNotExist {
+					return err
```

Comment:
You may want to check the tests for the method too. If I am right, then some test should have broken. If I misunderstood something, however, the tests should prove you right. So it's probably a good idea to double check our test coverage in this regards.